src/main/java/com/g4s8/ghman/web/GHAuthRedirectionWrap.java

```diff
@@ -0,0 +1,47 @@
+package com.g4s8.ghman.web;
+
+import java.io.IOException;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.cactoos.scalar.IoChecked;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.Take;
+import org.takes.facets.fork.FkWrap;
+import org.takes.facets.fork.Fork;
+import org.takes.misc.Opt;
+import org.takes.rs.RsRedirect;
+
+/**
+ * Github Authorization Redirection
+ * @since 1.0
+ * @todo #1:30min Refactor environment usage like APP_HOST, GH_CLIENT,
+ *  GH_SECRET, etc, in this class
+ *  and telegram classes, find a way to better configure these variables.
+ * @checkstyle LineLengthCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle ClassFanOutComplexityCheck (500 lines)
+ */
+public final class GHAuthRedirectionWrap extends FkWrap{
+
+    /**
+     * Ctor.
+     */
+    public GHAuthRedirectionWrap() {
+        super(
+            new Fork() {
+                Take take = req -> new RsRedirect(new IoChecked<>(() -> new URIBuilder("https://github.com/login/oauth/authorize")
+                .addParameter("redirect_uri", String.format("https://%s/auth", System.getenv("APP_HOST")))
```

Comment:
@samihibrahim please replace the call to static method `String.format` with the use of cactoos' `text.Formatted` class.